src/main/groovy/edu/oregonstate/mist/catalogapitest/db/CourseDAO.groovy

```diff
@@ -10,25 +11,79 @@ import org.skife.jdbi.v2.sqlobject.customizers.RegisterMapper
 @RegisterMapper(CourseMapper)
 public interface CourseDAO extends Closeable {
 
+    // Get all courses -------------------------------------------------------------------------------------------------
+
+    // GET
     @SqlQuery("""
             SELECT *
             FROM COURSES
             """)
     List<Course> getAllCourses()
 
+    // POST
+    @SqlUpdate("""INSERT INTO COURSES (CID, CRN, COURSENAME, INSTRUCTOR, DAY, TIME, LOCATION)
+                  values (COURSES_SEQ.NEXTVAL, :crn, :courseName, :instructor, :day, :time, :location)""")
+    void postCourse(@Bind("crn") Integer crn , @Bind("courseName") String courseName , @Bind("instructor") String instructor,
+                   @Bind("day") String day, @Bind("time") String time, @Bind("location") String location)
+
+    // CRN specific requests -------------------------------------------------------------------------------------------------
+
+    // GET
     @SqlQuery("""
             SELECT *
             FROM COURSES
             WHERE CRN = :crn
             """)
-    List<Course> findByCrn(@Bind("crn") Integer crn)
+    Course getByCrn(@Bind("crn") Integer crn)
+<<<<<<< HEAD
 
+    // Get specific instance number
+    @SqlQuery("""
+            SELECT COURSES_SEQ.CURRVAL FROM DUAL
```

Comment:
warning: getting the sequence number in a separate query may introduce race conditions. @adpextwindong has been looking into a similar issue. i'm not going to insist on code changes, but perhaps add a comment like `// FIXME: possible race condition`.